crates/adapters/src/transport/http/output.rs

```diff
@@ -150,15 +195,28 @@ impl HttpOutputEndpoint {
         let name = self.name().to_string();
         let guard = RequestGuard::new(finalizer);
 
+        let inner = self.inner.clone();
+
         HttpResponse::Ok()
             .insert_header(ContentType::json())
             .streaming(stream! {
                 let _guard = guard;
                 loop {
-                    match receiver.recv().await {
-                        Err(RecvError::Closed) => break,
-                        Err(RecvError::Lagged(_)) => (),
-                        Ok(buffer) => {
+                    // There is a bug in actix (https://github.com/actix/actix-web/issues/1313)
+                    // that prevents it from dropping HTTP connections on client disconnect
+                    // unless the endpoint periodically sends some data.  As a workaround,
+                    // if there is not real payload to send for more than 3 seconds, we will
+                    // generate an empty chunk.  Note that it takes 6s, i.e., 2x the timeout
```

Comment:
Is this timeout period internal to actix or something we can (or do) configure?

Reply:
I'm not super eager to add that control knob, and especially expose it to the client. Let's not do it now.